app/defines/interrupt_handler.h

```diff
@@ -35,30 +39,47 @@ void interrupt_handler() {
         }
 
     // execution of DIV, DIVU, REM, REMU
-    // The mcause gots reserved value from the spec
+    // The mcause gets reserved value from the spec
     if (mcause == CUSTOM_MCAUSE_DIV_OPERATION){
         csr_mtval = read_mtval();
         funct7    = extract_funct7(csr_mtval);
         funct3    = extract_funct3(csr_mtval);
+        // restore the rs1, rs2 values before entering the exception
```

Comment:
not sure what you mean by "restore"..
Did you mean "extract"?
Only the rd_addr needs to be "restored" -> but actually its more like a "override".